src/net.cpp

```diff
@@ -2635,25 +2635,34 @@ void CConnman::StopNodes()
         }
     }
 
-    // Close sockets
-    LOCK(cs_vNodes);
-    for (CNode* pnode : vNodes)
-        pnode->CloseSocketDisconnect();
-    for (ListenSocket& hListenSocket : vhListenSocket)
-        if (hListenSocket.socket != INVALID_SOCKET)
-            if (!CloseSocket(hListenSocket.socket))
-                LogPrintf("CloseSocket(hListenSocket) failed with error %s\n", NetworkErrorString(WSAGetLastError()));
+    // Delete peer connections
+    std::vector<CNode*> nodes;
+    {
+        LOCK(cs_vNodes);
+        nodes = std::move(vNodes);
+        vNodes.clear();
+    }
 
-    // clean up some globals (to help leak detection)
-    for (CNode* pnode : vNodes) {
+    for (CNode* pnode : nodes) {
+        pnode->CloseSocketDisconnect();
         DeleteNode(pnode);
     }
+
     for (CNode* pnode : vNodesDisconnected) {
         DeleteNode(pnode);
     }
-    vNodes.clear();
     vNodesDisconnected.clear();
+
+    // Close listening sockets
+    for (ListenSocket& hListenSocket : vhListenSocket) {
+        if (hListenSocket.socket != INVALID_SOCKET) {
+            if (!CloseSocket(hListenSocket.socket)) {
+                LogPrintf("CloseSocket(hListenSocket) failed with error %s\n", NetworkErrorString(WSAGetLastError()));
+            }
+        }
+    }
```

Comment:
minor nit: would be nice to do style-fixups in a separate commit

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/21563#issuecomment-824898734).